### PR TITLE
 fix(admin-tools): delete data which we store in options table when running data tools #3601 

### LINKED ISSUE #3601: fix(admin-tools): delete data which we stores in options table when run data tools
## User Story
As an `admin`, I want to remove extra data from the `wp_options` table.

When admin uses tool to recalculate stats or etc then plugin store few options for future reference which may not require when process complete, so it is fine to delete them instead of keeping them in the options table.
For example:
`give_recount_total_{form_id}`,
`give_recount_earnings_total`,
`give_temp_form_ids`,
`give_temp_all_payments_data`,
`give_temp_payment_items`,
`give_recount_all_total`
etc.

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix #3601 

## How Has This Been Tested?
Tested processing the diff option from the dropdown and then checking for the key in DB

## Screenshots (jpeg or gifs if applicable):
Video Link: https://screencast-o-matic.com/watch/cFjFFtqb3q

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by raftaar1191** [comment]
Slack Chat: https://givewp.slack.com/archives/C0FAGC83C/p1535110094000100

File: includes/admin/upgrades/upgrade-functions.php [hunk 1]
```diff
@@ -2758,6 +2762,48 @@ function give_v221_upgrades() {
...
+	$options_key = '\'' . implode( "','", $options_key ) . '\'';
+
+	global $wpdb;
+
+	/**
+	 * Update  1
+	 *
+	 * Change column length
+	 */
+	$wpdb->query( "DELETE FROM $wpdb->options WHERE option_name IN ( {$options_key} )" );
```
Comment (ravinderk) [inline includes/admin/upgrades/upgrade-functions.php]: @raftaar1191 `IN` query will be slow, so use multiple `option_name ={$option_name}` query

Comment (raftaar1191) [reply includes/admin/upgrades/upgrade-functions.php]: @ravinderk fix now the query will be like

```
string(920) "DELETE FROM wp_options WHERE option_name = 'give_temp_delete_form_ids' OR option_name = 'give_temp_delete_donation_ids' OR option_name = 'give_temp_delete_step' OR option_name = 'give_temp_delete_donor_ids' OR option_name = 'give_temp_delete_step_on' OR option_name = 'give_temp_delete_donation_ids' OR option_name = 'give_temp_delete_donor_ids' OR option_name = 'give_temp_delete_step' OR option_name = 'give_temp_delete_step_on' OR option_name = 'give_temp_delete_test_ids' OR option_name = 'give_temp_all_payments_data' OR option_name = 'give_recount_all_total' OR option_name = 'give_temp_recount_all_stats' OR option_name = 'give_temp_payment_items' OR option_name = 'give_temp_form_ids' OR option_name = 'give_temp_processed_payments' OR option_name = 'give_temp_recount_form_stats' OR option_name = 'give_temp_recount_earnings' OR option_name = 'give_recount_earnings_total' OR option_name = 'give_temp_reset_ids'"

```

Comment (ravinderk) [reply includes/admin/upgrades/upgrade-functions.php]: @raftaar1191 It is still the same as `IN` query. I will take care fo that

**Pull request merged**